### DISCUSSION TITLE
Getting access to entire Error object

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#3682] [General]
Hello,
We are using native fetch to fetch data from server and folowed the online example for error handling wheer we throuw an Error when we get anytihgn different from 200.

Problem is that we want access inside the onError to the entire json response that comes from serevr is there a way to achive that ?

**Comment by TkDodo** [comment]
You can reject a Promise with whatever content you like and that content will be available in the `error` field.

**Reply by Avid-Gholizadeh** [reply]
that is not working man.
when you use `Promise.reject({ //some key value here })` in interceptor,
the onError callback of tanstack query has an error argument that when you console this error it is like : 
Error: [object Object]

the onError function of react query does not access the error object.

**Reply by TkDodo** [reply]
it is working. show a reproduction where it isn’t please.
